src/Mesh/babylon.geometry.ts

```diff
@@ -233,12 +237,11 @@
             this._vertexBuffers[kind] = buffer;
 
             if (kind === VertexBuffer.PositionKind) {
-                var data = <FloatArray>buffer.getData();
-                var stride = buffer.getStrideSize();
-
-                this._totalVertices = data.length / stride;
+                if (totalVertices != null) {
+                    this._totalVertices = totalVertices;

```

Comment:
This should be documented as a **breaking change**.

If you set the vertex data with `setVerticesBuffer` rather than `setVerticesData` it now doesn't set `this._totalVertices` which means subMeshes don't get created; which means nothing draws.

Fix is to explicitly pass `totalVertices` into `setVerticesBuffer` when the vertex buffer is a `PositionKind` buffer; or pass `totalVertices` in for the `setIndices` call.

Reply:
This isn't supposed to be a breaking change. I must have done something unintentional. Let me investigate.

Reply:
Fix https://github.com/BabylonJS/Babylon.js/pull/4067

Reply:
```
                if (totalVertices == null) {
                    var data = buffer.getData()!;
                    var dataByteLength = (data instanceof Array) ? data.length * 4 : data.byteLength;
                    var bufferByteLength = dataByteLength - buffer.byteOffset;
                    this._totalVertices = bufferByteLength / buffer.byteStride;
                }
                else {
                    this._totalVertices = totalVertices;
                }
```

That's what I was going to do, but looks like you guys are on top of it.